content/sensu-go/5.17/api/secrets.md

```diff
@@ -0,0 +1,386 @@
+version: "5.17"
+product: "Sensu Go"
+menu:
+  sensu-go-5.17:
+    parent: api
+---
+
+- [The `/providers` endpoint](#the-providers-endpoint)
+  - [`/providers` (GET)](#providers-get)
+- [The `/providers/:provider` endpoint](#the-providers-provider-endpoint)
+  - [`/providers/:provider` (GET)](#providers-provider-get)
+  - [`/providers/:provider` (PUT)](#providers-provider-put)
+  - [`/providers/:provider` (DELETE)](#providers-provider-delete)
+- [The `/secrets` endpoint](#the-secrets-endpoint)
+  - [`/secrets` (GET)](#secrets-get)
+- [The `/secrets/:secret` endpoint](#the-secrets-secret-endpoint)
+  - [`/secrets/:secret` (GET)](#secrets-secret-get)
+  - [`/secrets/:secret` (PUT)](#secrets-secret-put)
+  - [`/secrets/:secret` (DELETE)](#secrets-secret-delete)
+
+**COMMERCIAL FEATURE**: Access secrets management in the packaged Sensu Go distribution.
+For more information, see [Get started with commercial features][1].
+
+## The `/providers` endpoint
+
+### `/providers` (GET)
+
+The `/providers` API endpoint provides HTTP GET access to a list of secrets providers.
+
+#### EXAMPLE {#providers-get-example}
+
+The following example demonstrates a request to the `/providers` API endpoint, resulting in a list of secrets providers.
+
+_**NOTE**: The `Env` provider is a built-in secrets provider that can retrieve backend environment variables as secrets._
+
+{{< highlight shell >}}
+curl -X GET \
+http://127.0.0.1:8080/api/enterprise/secrets/v1/providers \
+-H "Authorization: Bearer $SENSU_ACCESS_TOKEN"
+[
+  {
+    "type": "Env",
+    "api_version": "secrets/v1",
+    "metadata": {
+      "name": "env"
+    },
+    "spec": {
+    }
+  }
+]
+{{< /highlight >}}
+
+#### API Specification {#providers-get-specification}
+
+/providers (GET)  | 
+---------------|------
+description    | Returns the list of secret providers.
+example url    | http://hostname:8080/api/enterprise/secrets/v1/providers
+response type  | Array
+response codes | <ul><li>**Success**: 200 (OK)</li><li>**Error**: 500 (Internal Server Error)</li></ul>
+output         | {{< highlight shell >}}
+[
+  {
+    "type": "Env",
+    "api_version": "secrets/v1",
+    "metadata": {
+      "name": "env"
+    },
+    "spec": {
+    }
+  }
+]
+{{< /highlight >}}
+
+## The `/providers/:provider` API endpoint {#the-providers-provider-endpoint}
+
+### `/providers/:provider` (GET) {#providers-provider-get}
+
+The `/providers/:provider` API endpoint provides HTTP GET access to data for a specific `:provider`, by provider name.
+
+#### EXAMPLE {#providers-provider-get-example}
+
+In the following example, querying the `/providers/:provider` API endpoint returns a JSON map that contains the requested `:provider`.
+
+{{< highlight shell >}}
+curl -X GET \
+http://127.0.0.1:8080/api/enterprise/secrets/v1/providers/env \
+-H "Authorization: Bearer $SENSU_ACCESS_TOKEN"
+[
+  {
+    "type": "Env",
+    "api_version": "secrets/v1",
+    "metadata": {
+      "name": "env"
+    },
+    "spec": {
+    }
+  }
+]
```

Comment:
When getting a single provider `/providers/:provider` it should be a single JSON object, not a JSON list of objects.
```
{
    "type": "Env",
    "api_version": "secrets/v1",
    "metadata": {
        "name": "env"
    },
    "spec": {}
}
```